Updated common IE pipeline infer-request section (#10844)
openvinotoolkit/openvino: docs/OV_Runtime_UG/ov_infer_request.md

### Synchronous mode

> You can use `ov::InferRequest::infer`, which blocks the application execution, to infer model in synchronous mode:

Reviewer: ```suggestion
You can use `ov::InferRequest::infer`, which blocks the application execution, to infer model in the synchronous mode:
```

### Asynchronous mode

> Asynchronous mode can improve overall frame-rate of the application, because rather than wait for inference to complete, the app can continue doing things on the host, while accelerator is busy. You can use `ov::InferRequest::start_async` to infer model in asynchronous mode:

Reviewer: ```suggestion
Asynchronous mode can improve application's overall frame-rate, because rather than wait for inference to complete, the app can keep working on the host, while the accelerator is busy. You can use `ov::InferRequest::start_async` to infer model in the asynchronous mode:
```

@endsphinxdirective

> Asynchronous mode supports two ways to wait inference results:

Reviewer: ```suggestion
Asynchronous mode supports two ways the application waits for inference results:
```

>   * `ov::InferRequest::wait_for` - specifies maximum duration in milliseconds to block for. The method is blocked until the specified timeout has elapsed, or the result becomes available, whichever comes first.

Reviewer: ```suggestion
  * `ov::InferRequest::wait_for` - specifies the maximum duration in milliseconds to block the method. The method is blocked until the specified time has passed, or the result becomes available, whichever comes first.
```

    @endsphinxdirective

Both methods are thread-safe.

> When you are running several inference requests in parallel, a device can process them simultaneously, with no garauntees on the completion order. This may complicate a possible logic based on the `ov::InferRequest::wait` (unless you code needs to wait for the _all_ requests). For multi-requests scenario, consider using `ov::InferRequest::set_callback` method to set a callback which is  called upon completion of the request:

Reviewer: ```suggestion
When you are running several inference requests in parallel, a device can process them simultaneously, with no garauntees on the completion order. This may complicate a possible logic based on the `ov::InferRequest::wait` (unless your code needs to wait for the _all_ requests). For multi-request scenarios, consider using the `ov::InferRequest::set_callback` method to set a callback which is  called upon completion of the request:
```

@endsphinxdirective

> **NOTE**: Use weak reference of infer_request (`ov::InferRequest*`, `ov::InferRequest&`, `std::weal_ptr<ov::InferRequest>` and etc) in the callback, it is needed to avoid cyclic references.
For more details please take a look too [Classification Sample Async](../../samples/cpp/classification_sample_async/README.md).

> You can use `ov::InferRequest::cancel` method in case if you want to abort execution of the current inference request:

Reviewer 2: this works only on the CPU afaik... @vladimir-paramuzov ?

Author: but for other plugins it does nothing (no "not implemented" exception is thrown)

> > **NOTE**: Use weak reference of infer_request (`ov::InferRequest*`, `ov::InferRequest&`, `std::weal_ptr<ov::InferRequest>` and etc) in the callback, it is needed to avoid cyclic references.
> For more details please take a look too [Classification Sample Async](../../samples/cpp/classification_sample_async/README.md).
> 
> You can use `ov::InferRequest::cancel` method in case if you want to abort execution of the current inference request:

Reviewer: ```suggestion
> **NOTE**: Use weak reference of infer_request (`ov::InferRequest*`, `ov::InferRequest&`, `std::weal_ptr<ov::InferRequest>`, etc.) in the callback. It is necessary to avoid cyclic references.
For more details, check [Classification Sample Async](../../samples/cpp/classification_sample_async/README.md).

You can use the `ov::InferRequest::cancel` method if you want to abort execution of the current inference request:
```

### Using of ROI tensors

> It is possible to re-use shared input by several models. You do not need to allocate separate input tensor for a model if it processes a ROI object located inside of already allocated input of a previous model. For instance, when first model detects objects on a video frame (stored as input tensor) and second model accepts detected bounding boxes (ROI inside of the frame) as input. In this case, it is allowed to re-use pre-allocated input tensor (used by first model) by second model and just crop ROI without allocation of new memory using `ov::Tensor` with passing of `ov::Tensor` and `ov::Coordinate` as parameters.

Reviewer: ```suggestion
It is possible to re-use shared input by several models. You do not need to allocate separate input tensor for a model if it processes a ROI object located inside of already allocated input of a previous model. For instance, when the first model detects objects in a video frame (stored as input tensor) and the second model accepts detected bounding boxes (ROI inside of the frame) as input. In this case, it is allowed to re-use pre-allocated input tensor (used by the first model) by the second model and just crop ROI without allocation of new memory using `ov::Tensor` with passing of `ov::Tensor` and `ov::Coordinate` as parameters.
```
